core/vm/contract.go

```diff
@@ -148,11 +150,16 @@ func (c *Contract) GetOp(n uint64) OpCode {
 
 // GetByte returns the n'th byte in the contract's byte array
 func (c *Contract) GetByte(n uint64) byte {
-	if n < uint64(len(c.Code)) {
+	if n < c.CodeEndOffset() {
 		return c.Code[n]
 	}
 
-	return 0
+	// For legacy contracts running out of code section means STOP
+	if c.header.codeSize == 0 {
+		return 0
+	}
+	// For EOF contracts running out of code section means abort execution with failure
+	return 0xfe
```

Comment:
Nice trick.

Reply:
However isn't there an enum for opcodes? If so, it would make sense using it for both `0` and `0xfe` so it is more apparent to the reader.

Reply:
> However isn't there an enum for opcodes? If so, it would make sense using it for both 0 and 0xfe so it is more apparent to the reader.

No enum for `0xfe` yet :smiley: Also this function returns `byte` that's why it used `0` I think.
However it doesn't make sense to return `byte` because it's always casted to `OpCode` in `GetOp` above.
I guess I'll submit a separate small cleanup PR for this.

Reply:
For anyone interested, see #23974 and #24017.